### ISSUE TITLE
Rule request: Prefer enums for convenience types

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1871]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

If a type only exists to group global (`static`) functions and variables, it should be an `enum` to avoid being instantiated.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

From [Erica Sadun's Swift Style book](https://pragprog.com/book/esswift/swift-style):

> When creating new types to host global constants and functions, prefer caseless enumerations to using structs. Enums are the conventional approach used by the Swift core team. Unlike caseless enumerations, structs can be instantiated by an API client.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// should trigger
struct Math { 
  public static let pi = 3.14
}

class Math { 
  public static let pi = 3.14
}

struct Math {
  public static let pi = 3.14 
  @available(*, unavailable) init() {}
}

// shouldn't trigger

// enum
enum Math {
  public static let pi = 3.14
}

// class visible to Objective-C
class Math: NSObject {
  public static let pi = 3.14
}

// type with non-static declarations
struct Math {
  public static let pi = 3.14
  public let randomNumber = 2
}
```
3. Should the rule be configurable, if so what parameters should be configurable?

Just on the severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default.
